### ISSUE TITLE
Other ships don't make engine sounds

### ISSUE THREAD

**Issue by Pointedstick** [endless-sky/endless-sky#522]
I suspect this is to prevent too many noises from going on at once and creating an auditory mess, but you do hear their weapons. Might wanna try it out.

**Comment by guillochon** [comment]
I like this idea, could just make them no more than ~20% the volume at maximum, and less than that if they are further.

**Comment by endless-sky** [comment]
If you're building from the latest source, give it a try and see what you think: 4c31f9d. I do like how it means that an empty system is now totally quiet (except the noises from your own ship) and a busy system ahs a lot of background noise, but I wonder if it would get annoying after a while.

**Comment by Pointedstick** [comment]
Oh, that's great. I definitely prefer it this way. Makes the other ships feel more "alive" and gives the whole game a bit more of a soundscape.

**Comment by guillochon** [comment]
Yeah definitely a nice improvement.

**Comment by egilburg** [comment]
I wonder if it would be computationally feasible to make the sounds doppler-shiftable? As NPCs zoom towards/away from you, the frequency slightly changes for that "vrooom" sound, and the faster they move past you the sharper the transition is.

**Comment by endless-sky** [comment]
OpenAL does support Doppler shift, but one way I'm optimizing the sounds is that if the same sound is playing from more than two sources, I play only one copy of the sound, but at a higher volume, rather than two distinct copies. (So, two sounds playing at the same distance get merged into one sound at that distance divided by sqrt(2).) The "merged" source is also positioned so that the left-right balance matches the weighted average of all the sounds that were merged to create it.

In theory I could do the same with the Doppler shift, as well - take a weighted average of the relative velocities of the sounds - but in practice I wasn't able to get a Doppler effect that sounded realistic, and I was worried that it might be one of the reasons the sounds are so much louder or softer on certain systems, so I disabled it.
